app/models/activity.rb

```diff
@@ -115,6 +115,11 @@ def student_only
       subject_area_list
     end
     integer :project_ids, :multiple => true, :references => Admin::Project
+
+    string :project_names, :multiple => true do
+      projects.map { |p| p.name }
+    end
+
```

Comment:
I added these for the related resources query:

<pre>
                fulltext "*" do
                    boost(4.0) { with(:subject_areas, tags['subject_areas']) }
                    boost(2.0) { with(:grade_levels, tags['grade_levels']) }
                    **boost(1.0) { with(:project_names, projects) }**
                end

</pre>